src/actions/dataset.ts

```diff
@@ -48,4 +49,23 @@ export function datasetUrlLoad(name: string, url: string): DatasetUrlLoad {
       }
     );
   };
-}
+};
+
+export const DATASET_RECEIVE = 'DATASET_RECEIVE';
```

Comment:
Thanks.  These are great questions.  :) 

> It has not been clear to me what the app does with either the URL or the original data (in the case of inline data), it puts it in the store

Currently, we just stick the `data` in the rendered specifications. 
So for UrlData, Vega runtime will read from the url again to render each plot. 
For InlineData, Vega just reads the data inline.  

Eventually, we would change both cases to bind data at runtime so that we can reduce redundant transformation.  Thinking more about this, one of the first place to apply common transformation would be filtering UI.  (We don't have filter UI yet in this new version though). 

> i was under the impression that the schema also encapsulates the data?

The schema reads the data to create a schema info (describing each field in the dataset) and calculate basic statistics, but the schema object does not store the data.  :)